jc-1721-a connectors gs cleanup (#616)
redhat-developer/app-services-guides: docs/connectors/getting-started-connectors/README.adoc

For Service Preview, you have two choices:

* *The hosted preview environment*

** The {connectors} instances are hosted on a multitenant {osd-name-short} cluster that is owned by Red Hat.
** You can create four {connectors} instances at a time.

> ** The preview environment applies 48-hour expiration windows, as described in https://access.redhat.com/documentation/en-us/openshift_connectors/1/guide/8190dc9e-249c-4207-bd69-096e5dd5bc64[Red Hat {openshift} {connectors} Service Preview guidelines^].

Reviewer: The referenced doc still contains "evaluation" in its title:
https://access.redhat.com/documentation/en-us/openshift_connectors/1/guide/8190dc9e-249c-4207-bd69-096e5dd5bc64

> ** The preview environment applies 48-hour expiration windows, as described in https://access.redhat.com/documentation/en-us/openshift_connectors/1/guide/8190dc9e-249c-4207-bd69-096e5dd5bc64[Red Hat {openshift} {connectors} Service Preview guidelines^].

Author: yes - it does - I will change it

* *Your own trial environment*

** You have access to your own {osd-name-short} trial environment.
** You can create an unlimited number of {connectors} instances.
** Your {osd-name-short} trial cluster expires after 60 days.
** A cluster administrator has installed the {product-connectors} add-on as described in https://access.redhat.com/documentation/en-us/openshift_connectors/1/guide/15a79de0-8827-4bf1-b445-8e3b3eef7b01[Adding the Red Hat {openshift} {connectors} add-on to your {osd-name-short} trial cluster^].

*Configuring {product-long-kafka} for use with {product-connectors}*

ifndef::qs[]
Complete the steps in {base-url}{getting-started-url-kafka}[Getting started with {product-long-kafka}^] to set up the following components:
endif::[]

> ifdef::qs[]

Author: Yes - I agree

Make sure that you have set up the prerequisite components.

.Verification
* Is the Kafka instance listed on the *Kafka Instances* page and is the Kafka instance in the *Ready* state?
* Is your service account created on the *Service Accounts* page?
* Did you save your service account credentials to a secure location?
* Are the permissions for your service account listed on the *Access* page of the Kafka instance?

> * Is the Kafka topic that you created for {connectors} listed on the page for the Kafka instance?

Author: there is no page title - however, there is a **Kafka Instances** page - which is why I worded it this way - to hopefully avoid confusion

* If you plan to use a 60-day {osd-name-short} trial cluster to deploy your {product-connectors} instances, has a cluster administrator added the {product-connectors} add-on to your trial cluster?

endif::[]

ifndef::qs[]
.Verification
* Verify that the Kafka instance is listed on the *Kafka Instances* page and that the state of the Kafka instance is shown as *Ready*.
* Verify that your service account was successfully created on the *Service Accounts* page.
* Verify that you saved your service account credentials to a secure location.
* Verify that the permissions for your service account are listed on the *Access* page of the Kafka instance.

> * Verify that the Kafka topic that you created for {product-connectors} is listed on the page for the Kafka instance.

Author: there is no page title - however, there is a **Kafka Instances** page - which is why I worded it this way - to hopefully avoid confusion

> * Verify that the Kafka topic that you created for {product-connectors} is listed on the page for the Kafka instance.

Reviewer: Same reply as for my previous comment, so I'll unresolve this conversation for now until we clarify :)

+
Click the card to select the connector, and then click *Next*.

. On the *Kafka instance* page, click the card for the {product-kafka} instance that you configured for {connectors}, and then click *Next*.

. On the *Namespace* page, the namespace that you select depends on your {osd-name-short} environment. The namespace is the deployment space that hosts your {connectors} instances.
+
If you're using a trial cluster in your own {osd-name-short} environment, select the card for the namespace that was created when a system administrator added the {connectors} service to your trial cluster, as described in https://access.redhat.com/documentation/en-us/openshift_connectors/1/guide/15a79de0-8827-4bf1-b445-8e3b3eef7b01[Adding the Red Hat {openshift} {connectors} add-on to your {osd-name-short} trial cluster^].
+
If you're using the hosted preview environment, click *Create preview namespace* to provision a namespace for hosting the {connectors} instances that you create.

. Click *Next*.

. Configure the core configuration for your {connectors} instance:

> .. Type a name for your {connectors} instance. For example, type *hello world generator*.

Author: they do not differ

.. Type the *Client ID* and *Client Secret* of the service account that you created for {connectors} and then click *Next*.
. Provide connector-specific configuration. For the *Data Generator*, provide the following information:
.. *Topic Name*: Type the name of the Kafka topic that you created for {connectors}. For example, type *test-topic*.
.. *Content Type*: Accept the default, `text/plain`.
.. *Message*: Type the content of the message that you want the {connectors} instance to send to the Kafka topic. For example, type `Hello World!!`.
.. *Period*: Specify the interval (in milliseconds) at which you want the {connectors} instance to send messages to the Kafka topic. For example, specify `10000`, to send a message every 10 seconds.
.. *Data Shape Produces Format*: Accept the default, `application/octet-stream`.

. Click *Next*.

> . Select an error handling policy for your {connectors} instance. The options are:

Reviewer: Suggested two possible rewrites for this text.

> . Select an error handling policy for your {connectors} instance. The options are:

Reviewer: The updated text now reads "one of the following error handling policy". 
We should change "policy" to "policies"?

[role="_abstract"]

> A *sink* connector consumes messages from a Kafka topic and sends them to an external system.

Reviewer: ```suggestion
A _sink_ connector consumes messages from a Kafka topic and sends them to an external system.
```

> For this example, you use the *HTTP Sink* connector which consumes the Kafka messages (produced by your Data Generator source {connectors} instance) and sends the messages to an HTTP endpoint.

Reviewer: ```suggestion
For this example, you use the _HTTP Sink_ connector which consumes the Kafka messages (produced by your Data Generator source {connectors} instance) and sends the messages to an HTTP endpoint.
```

Reviewer: Maybe remove the bold highlighting for the connector name entirely (rather than put it in italics as I'd previously suggested), for consistency with the (non-) highlighting of  "Data Generator source connector" in the previous section?

.Prerequisites

ifndef::qs[]
* You're logged in to the {product-long-connectors} web console at {service-url-connectors}[^].
endif::[]

> * You created the source {connectors} instance as described in _Creating a {connectors} instance for a data source_.

Reviewer: Include a hyperlink to the referenced section?

* For the data sink example, open the free https://webhook.site[webhook.site^] in a browser window. The `webhook.site` page provides a unique URL that you copy for use as an HTTP data sink.

.Procedure

> . In the {product-long-connectors} web console, click *Create {connectors} instance*.

Reviewer: ```suggestion
. In the {product-long-connectors} web console, click *Create a {connectors} instance*.
```
Also include a link to the Connectors web console.

Reviewer: This change has not been made? Need to insert "a" after "Create".

. Select the sink connector that you want to use:

> .. For example, type *http* in the search field. The list of {connectors} filters to show the *HTTP sink* connector.

Reviewer: ```suggestion
.. For example, type `http` in the search field. The list of {connectors} is filtered to show the *HTTP sink* connector.
```

.. Click the *HTTP sink* card and then click *Next*.

> . On the *Kafka instance* page, select the {product-kafka} instance for the connector to work with.

Reviewer: ```suggestion
. On the *Kafka Instance* page, select the {product-kafka} instance for the connector to work with.
```

+
For example, select *test* and then click *Next*.

. On the *Namespace* page, the namespace that you select depends on your {osd-name-short} environment. The namespace is the deployment space that hosts your {connectors} instances.
+
If you're using a trial cluster on your own {osd-name-short} environment, select the card for the namespace that was created when you added the {connectors} service to your trial cluster.
+

> If you're using the hosted preview environment, click the *preview namespace* that you provisioned when you created the source connector.

Reviewer: Similar comment as before: If the subsequent steps don't apply to both types of environment, clarify which environment they apply to.

Author: NPTF - the subsequent steps apply to both environments

. Click *Next*.

. Provide the core configuration for your connector:

> .. Type a unique name for the connector. For example, type *hello world receiver*.

Reviewer: ```suggestion
.. Type a unique name for the connector. For example, type `hello world receiver`.
```

> .. Type the *Client ID* and *Client Secret* of the service account that you created for {connectors} and then click *Next*.

Reviewer: ```suggestion
.. In the *Client ID* and *Client Secret* fields, type the credentials of the service account that you created for {connectors} and then click *Next*.
```

. Provide the connector-specific configuration for your *HTTP sink* {connectors} instance:

> .. *Topic Names*: Type the name of the topic that you used for the source {connectors} instance. For example, type *test-topic*.

Reviewer: ```suggestion
.. *Topic Names*: Type the name of the topic that you used for the source {connectors} instance. For example, type `test-topic`.
```

.. *Method*: Accept the default, `POST`.
.. *URL*: Type your unique URL from the link:https://webhook.site[webhook.site^].
.. *Data Shape Consumes Format*: Accept the default, `application/octet-stream`.

. Click *Next*.

. Select an error handling policy for your {connectors} instance. For example, select *Stop* and then click *Next*.

. Review the summary of the configuration properties and then click *Create {connectors} instance*.
+
Your {connectors} instance is added to the *{connectors} Instances* page.
+

> After a couple of seconds, the status of your {connectors} instance changes to the *Ready* state. It consumes messages from the associated Kafka topic and sends them to the data sink (for this example, the data sink is the HTTP URL that you provided).

Reviewer: Verify that you are using a consistent style for states such as Ready.
